test/metabase/driver/sql/query_processor_test.clj

```diff
@@ -465,3 +470,26 @@
                               :alias        "CategoriesStats"
                               :fields       :all}]
                :limit       3}))))))
+
+(deftest expressions-and-coercions-test
+  (mt/test-drivers (conj (sql-jdbc.tu/sql-jdbc-drivers) :bigquery)
+    (testing "Don't cast in both inner select and outer select when expression (#12430)"
+      (mt/with-temp-copy-of-db
+        (let [price-field-id (data/id :venues :price)]
+          (db/update! Field price-field-id {:coercion_strategy :Coercion/UNIXSeconds->DateTime
+                                            :effective_type    :type/DateTime})
```

Comment:
You should use `mt/id` here instead of `data/id` -- `mt/id` is just an alias for `data/id`, but it's better to use the `mt/` versions since it collects all the test utils in one place instead of requiring people to remember like 10 different test util namespaces. When other people look at this test for an example for how to write the next one, I'd rather them see `mt/id` than `data/id`.

Also.. you can do `mt/with-temp-vals-in-db` instead of `mt/with-temp-copy-of-db` + `db/update!`. It has the added benefit of including `testing` context when you use it